CraftingRevisions/Patches/GameManager_Awake.cs

```diff
@@ -32,13 +32,29 @@ private static void Postfix(Il2CppTLD.Gear.BlueprintManager __instance)
 					// load the blueprint into the game
 					bool loaded = __instance.LoadUserBlueprint(jsonUserBlueprint);
 
-#warning TODO - there appears to be no way to define an icon for this blueprint (UserBlueprintData still a WIP?)
-#warning TODO - get CraftingAudio working, is a string but unclear what value it wants. (see UserBlueprintData.MakeRuntimeWwiseEvent)
+					if (loaded == false)
+					{
+						// implement validation here
 #warning TODO - work on loading validation if it fails
-#warning TODO - Should we attempt to detect & convert "old" format json ?
 
+					}
 				}
 			}
 		}
 	}
+
+	// So these patches are to work around the fact that "UserBlueprintData.MakeRuntimeWwiseEvent" appears to be broken or unfinished
+	[HarmonyPatch(typeof(Il2CppTLD.Gear.UserBlueprintData), nameof(Il2CppTLD.Gear.UserBlueprintData.MakeRuntimeWwiseEvent), new Type[] { typeof(string) })]
+	internal class UserBlueprintData_MakeRuntimeWwiseEvent
+	{
+		private static void Prefix(string eventName, ref bool __runOriginal)
+		{
+			__runOriginal = false;
+		}
+		private static void Postfix(string eventName, ref bool __runOriginal, ref Il2CppAK.Wwise.Event? __result)
+		{
+			__result = BlueprintManager.MakeAudioEvent(eventName);
```

Comment:
Move MakeAudioEvent to this class. I don't see any reason for it to be exposed to the rest of the mod.